### ISSUE TITLE
Bulk update changes

### ISSUE THREAD

**Issue by lucko** [lucko/LuckPerms#1836]
A bulk list of ideas for bulk update!

To (potentially) include:

- [Filter by temporary permission](https://github.com/lucko/LuckPerms/issues/1350)
- [Dry run argument](https://github.com/lucko/LuckPerms/issues/1037)
- [Auto remove groups after deletion](https://github.com/lucko/LuckPerms/issues/1249)

**Comment by BrainStone** [comment]
I don't think it makes sense throwing the auto remove groups after deletion here.

**Comment by Turbotailz** [comment]
> I don't think it makes sense throwing the auto remove groups after deletion here.

Why not? Seems like the perfect execution for such a task, IMO. At least make it optional, like a prompt message after the command has been run to delete any groups.

**Comment by BrainStone** [comment]
I still feel like the last two are best sloved by a flag (or a flag to not do it) and so I don't quite understand why they're being grouped here.

**Comment by Andre601** [comment]
> I still feel like the last two are best sloved by a flag (or a flag to not do it) and so I don't quite understand why they're being grouped here.

I personally think that this wouldn't make sense.
The rename command already has the disadvantage, that it just renames the group itself, but not any appearance of `group.<group>` in other places, making people lose their ranks or make groups lose their inheritance.
So at least having the rename command perform the bulk edit action would be a QOL improvement by removing an additional step of a potentially dangerous command, that not everyone understands...
Because basically is the rename command trash as it doesn't do anything useful and the user has to make the bulk edit anyway...

**Comment by Tobi406** [comment]
I'd say

- Add a message that x happened (eg. "All permissions called group.x have been removed")
- And sure a flag/argument to not have it happening at all

**Comment by BrainStone** [comment]
I'll post it again:

For rename: Have it bulkupdate by default and a flag to disable that behavior. So that the most common use case is the default.

**Comment by lucko** [comment]
To also include #2362

**Comment by lucko** [comment]
#2484

**Comment by SrBedrock** [comment]
Add possibility to modify/remove `meta` through bulkupdate?
Example: `/lp bulkupdate users delete "meta == 14_12"`

**Comment by Fefo6644** [comment]
> Add possibility to modify/remove `meta` through bulkupdate?
> Example: `/lp bulkupdate users delete "meta == 14_12"`

That is possible with the current setup, meta nodes are structured like so `meta.<key>.<value>`

**Comment by Laarryy** [comment]
Adding #4026 to this - would be nice to see a bulkupdate way to select and modify perms more permissively
